source/Engine.cpp

```diff
@@ -2211,9 +2211,26 @@ void Engine::DoCollection(Flotsam &flotsam)
 			commodity = outfit->DisplayName();
 			player.Harvest(outfit);
 		}
-		else
+		else {
+			string displayName = outfit->DisplayName();
+			string pluralName = outfit->PluralName();
+
+			if(!player.OutfitIsKnown(*outfit))
+			{
+				displayName = "Unknown " + outfit->Category();
+
+				if((amount == 1) && (displayName.back() == 's'))
+				{
+					displayName.pop_back();
+					displayName += " Component";
+				}
+
+				pluralName = displayName + 's';
```

Comment:
I see this kind of logic at multiple places, how about making it an outfit method?

Reply:
I need to see how to refractor ir

Reply:
@Hurleveur, is there a global class which I can use to put the function at?

Reply:
Cant you just add that to outfits? you can even make it a method that doesnt require an instance (how is it called again? static I think)

Reply:
It shouldnt be a global thing I think since its only used for outfits

Reply:
are you suggesting to add a func to the outfit class that will generate the obscure name?

Reply:
Yeah exactly
UnknownName()

Reply:
seems fair, I just need to see if the plurals will fit in nicelly

Reply:
Just give the amount there is as an argument to the method